rust/arrow/src/compute/kernels/comparison.rs

```diff
@@ -22,17 +22,50 @@
 //! `RUSTFLAGS="-C target-feature=+avx2"` for example.  See the documentation
 //! [here](https://doc.rust-lang.org/stable/core/arch/) for more information.
 
+use regex::Regex;
+use std::collections::HashMap;
 use std::sync::Arc;
 
 use crate::array::*;
-use crate::buffer::Buffer;
 use crate::compute::util::apply_bin_op_to_option_bitmap;
-use crate::datatypes::{ArrowNumericType, BooleanType, DataType, ToByteSlice};
+use crate::datatypes::{ArrowNumericType, BooleanType, DataType};
 use crate::error::{ArrowError, Result};
 
 /// Helper function to perform boolean lambda function on values from two arrays, this
 /// version does not attempt to use SIMD.
-pub fn compare_op<T, F>(
+macro_rules! compare_op {
+    ($left: expr, $right:expr, $op:expr) => {{
+        if $left.len() != $right.len() {
+            return Err(ArrowError::ComputeError(
+                "Cannot perform math operation on arrays of different length".to_string(),
```

Comment:
This error message refers to "math operation" which seems incorrect?

Reply:
Looks like the original comparisons.rs was referring to "math operation" as well: 

https://github.com/apache/arrow/blob/master/rust/arrow/src/compute/kernels/comparison.rs#L46

But I can definitely fix that!